.github/workflows/docker-build.yml

```diff
@@ -284,7 +292,7 @@ jobs:
       - name: Check out the code at a specific ref
         uses: actions/checkout@v4
         with:
-          ref: ${{ needs.get-version.outputs.version || github.ref }}
+          ref: ${{ inputs.nightly_tag_main || github.ref }}
```

Comment:
Simplification were mainly around this. We need a separate `ref` to allow you to specify the branch/commit/etc to check out in case that's different than what we want the image tagged as.

Reply:
love that, but trying to keep scope under control. nightlies have been broken for weeks